administrator/components/com_associations/views/association/view.html.php

```diff
@@ -147,11 +147,28 @@ protected function addToolbar()
 		$input = JFactory::getApplication()->input;
 		$input->set('hidemainmenu', 1);
 
+		$user       = JFactory::getUser();
+		$userId     = $user->id;
+		$checkedOut = !($this->table->checked_out == 0 || $this->table->checked_out == $userId);
+		$component  = $input->get('acomponent', '', 'string');
+
 		JToolbarHelper::title(JText::_('COM_ASSOCIATIONS_HEADER_EDIT'), 'contract');
 
-		JToolbarHelper::apply('reference', 'COM_ASSOCIATIONS_SAVE_REFERENCE');
+		$canEdit = $user->authorise('core.edit', $component . $this->referenceId);
+		if (isset($this->table->created_by))
+		{
+			$canEditOwn = $user->authorise('core.edit.own', $component . $this->referenceId) && $this->table->created_by == $userId;
+		}
+
```

Comment:
Is this way of getting the authorise? I'm using this since it doesn't need to load a helper like in:
https://github.com/joomla-projects/gsoc16_improved-multi-lingual/blob/staging/administrator/components/com_contact/views/contact/view.html.php#L90

or

https://github.com/joomla-projects/gsoc16_improved-multi-lingual/blob/staging/administrator/components/com_menus/views/item/view.html.php#L57

If this is wrong please tell me @infograf768 @andrepereiradasilva @alikon